src/main/java/ch/njol/skript/expressions/ExprLore.java

```diff
@@ -1,236 +1,181 @@
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * TODO make a 'line %number% of %text%' expression and figure out how to deal with signs (4 lines, delete = empty, etc...)
- *
- * @author joeuguce99
  */
 @Name("Lore")
 @Description("An item's lore.")
-@Examples("set the 1st line of the item's lore to \"&lt;orange&gt;Excalibur 2.0\"")
+@Example("""
+	set {_item} to diamond named "Bling Diamond" with lore "&bThe blingiest of diamonds"
+	add "With extra steps" to lore of {_item}
+	remove line 1 of lore of {_item} from lore of {_item}
+	set line 3 of lore of {_item} to "-----"
+	""")
 @Since("2.1")
 public class ExprLore extends SimpleExpression<String> {
 
+	private static final boolean IS_RUNNING_1_21_5 = Skript.isRunningMinecraft(1,21,5);
+
 	static {
-			Skript.registerExpression(ExprLore.class, String.class, ExpressionType.PROPERTY,
-					"[the] lore of %itemstack/itemtype%", "%itemstack/itemtype%'[s] lore",
-					"[the] line %number% of [the] lore of %itemstack/itemtype%",
-					"[the] line %number% of %itemstack/itemtype%'[s] lore",
-					"[the] %number%(st|nd|rd|th) line of [the] lore of %itemstack/itemtype%",
-					"[the] %number%(st|nd|rd|th) line of %itemstack/itemtype%'[s] lore");
+		Skript.registerExpression(ExprLore.class, String.class, ExpressionType.PROPERTY,
+				"[the] lore of %itemstack/itemtype%",
+				"%itemstack/itemtype%'[s] lore",
+				"[the] line %number% of [the] lore of %itemstack/itemtype%",
+				"[the] line %number% of %itemstack/itemtype%'[s] lore",
+				"[the] %number%(st|nd|rd|th) line of [the] lore of %itemstack/itemtype%",
+				"[the] %number%(st|nd|rd|th) line of %itemstack/itemtype%'[s] lore");
 	}
 
-	@Nullable
-	private Expression<Number> lineNumber;
-
-	@SuppressWarnings("null")
+	private @Nullable Expression<Number> lineNumber;
 	private Expression<?> item;
 
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
 		lineNumber = exprs.length > 1 ? (Expression<Number>) exprs[0] : null;
 		item = exprs[exprs.length - 1];
 		return true;
 	}
 
 	@Override
-	@Nullable
-	protected String[] get(final Event e) {
-		final Object i = item.getSingle(e);
-		final Number n = lineNumber != null ? lineNumber.getSingle(e) : null;
-		if (n == null && lineNumber != null)
+	protected String @Nullable [] get(Event event) {
+		if (!validateItem(item.getSingle(event)))
+			return null;
+		ItemStack itemStack = ItemUtils.asItemStack(item.getSingle(event));
+		assert itemStack != null; // Validated in validateItem
+		ItemMeta itemMeta = itemStack.getItemMeta();
+		//noinspection deprecation
+ 		List<String> itemLore = itemMeta.getLore();
+		if (itemLore == null || itemLore.isEmpty())
 			return null;
```

Comment:
```suggestion
			return new String[0];
```

Reply:
Is skript wanting to start enforcing this as a convention? I'm not against it just seems like more behavior to handle where null would suffice

Reply:
Yep. Internally, `null` has to be handled by creating an array of 0 length dynamically. By returning an empty array here, we avoid that operation (which is more expensive)